fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVPlanUtil.java

```diff
@@ -150,4 +173,117 @@ public static Set<TableIf> getBaseTableFromQuery(String querySql, ConnectContext
             }
         }
     }
+
+    public static List<ColumnDefinition> generateColumnsBySql(String querySql, ConnectContext ctx, String partitionCol,
+            Set<String> distributionColumnNames, List<SimpleColumnDefinition> simpleColumnDefinitions,
+            Map<String, String> properties) {
+        List<StatementBase> statements;
+        try {
+            statements = new NereidsParser().parseSQL(querySql);
+        } catch (Exception e) {
+            throw new ParseException("Nereids parse failed. " + e.getMessage());
+        }
+        StatementBase parsedStmt = statements.get(0);
+        LogicalPlan logicalPlan = ((LogicalPlanAdapter) parsedStmt).getLogicalPlan();
+        StatementContext original = ctx.getStatementContext();
+        try (StatementContext tempCtx = new StatementContext()) {
+            ctx.setStatementContext(tempCtx);
+            try {
+                NereidsPlanner planner = new NereidsPlanner(ctx.getStatementContext());
+                Plan plan = planner.planWithLock(logicalPlan, PhysicalProperties.ANY, ExplainLevel.ANALYZED_PLAN);
+                return generateColumns(plan, ctx, partitionCol, distributionColumnNames, simpleColumnDefinitions,
+                        properties);
+            } finally {
+                ctx.setStatementContext(original);
+            }
+        }
+    }
+
+    public static List<ColumnDefinition> generateColumns(Plan plan, ConnectContext ctx, String partitionCol,
```

Comment:
add java doc style's comment to explain what this function want to do, the input paramters and the return value